packages/nightly/src/lib/nightly.ts

```diff
@@ -0,0 +1,175 @@
+import type {
+  InjectedWallet,
+  InjectedWalletMetadata,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from "@near-wallet-selector/core";
+import { createAction } from "@near-wallet-selector/wallet-utils";
+import { isMobile } from "is-mobile";
+import { utils } from "near-api-js";
+import { JsonRpcProvider } from "near-api-js/lib/providers";
+import { createTransaction } from "near-api-js/lib/transaction";
+import { Account, waitFor } from "@near-wallet-selector/core";
+import type { NearNightly, NightlyInjected } from "./injected-nightly";
+import { PublicKey } from "near-api-js/lib/utils";
+
+declare global {
+  interface Window {
+    nightly: NightlyInjected | undefined;
+  }
+}
+
+interface NightlyState {
+  wallet: NearNightly;
+}
+
+const setupNightlyState = (meta: InjectedWalletMetadata): NightlyState => {
+  const wallet = window.nightly?.near;
+  if (!wallet) {
+    window.location.href = meta.downloadUrl;
+    throw new Error("Redirecting to download");
+  }
+  return {
+    wallet: wallet,
+  };
+};
+
+const Nightly: WalletBehaviourFactory<InjectedWallet> = async ({
+  options,
+  store,
+  logger,
+  metadata,
+}) => {
+  const _state = setupNightlyState(metadata);
+  const provider = new JsonRpcProvider({ url: options.network.nodeUrl });
+  const state = store.getState();
+
+  if (state.selectedWalletId === "nightly") {
+    await _state.wallet.connect();
+  }
+  return {
+    // nightly does not support delegating signing right now
+    async signIn() {
+      const existingAccount = _state.wallet.account.accountId;
+
+      if (existingAccount) {
+        const nearAccount: Account = {
+          accountId: _state.wallet.account.accountId,
+        };
+        return [nearAccount];
+      }
+
+      const { accountId } = await _state.wallet.connect();
+      const nearAccount: Account = { accountId };
+      return [nearAccount];
+    },
+
+    async signOut() {
+      await _state.wallet.disconnect();
+    },
+
+    async getAccounts() {
+      const nearAccount: Account = {
+        accountId: _state.wallet.account.accountId,
+      };
+      return [nearAccount];
```

Comment:
We should return an empty array if there's no account (I think Nightly returns an empty string for `accountId` and a public key with all 1s) and use this method in `signIn` to avoid duplicate checking logic

Reply:
Fixed